heft-plugins/heft-lint-plugin/src/LintPlugin.ts

```diff
@@ -99,8 +110,42 @@ export default class LintPlugin implements IHeftTaskPlugin<ILintPluginOptions> {
               this._lintingPromises.push(lintingPromise);
             }
           );
+          // Set the flag to indicate that we are in the typescript phase
+          inTypescriptPhase = true;
         }
       );
+
+      if (!inTypescriptPhase) {
```

Comment:
If not executing in the typescript phase, this needs to happen in `runAsync`, because otherwise it might reference files that don't exist yet (e.g. generated files by other build tasks).

Reply:
There doesn't appear to be a `runAsync` hook on taskSession? Did you mean `run` thats further down below?

> The run hook is called after all dependency task executions have completed during a normal run, or during a watch mode run when no runIncremental hook is provided. It is where the plugin can perform its work. To use it, call run.tapPromise(<pluginName>, <callback>).

Reply:
Yep, run.